modules/jaggery-apps/portal/util/utility.jag

```diff
@@ -95,24 +35,19 @@ function setServerURL() {
             }
         }
 
-        //create server service url
-        serverUrl = identityServerURL.getServerURL("", true, true) + "/services";
-
-        //basic server core url : https://<HOST_NAME>:<PORT>/<PROXY_CONTEXT_PATH>
-        var serverCoreUrl = identityServerURL.getServerURL("", true, false);
+        if ( serviceOrigin == null || serviceOrigin === '' || serviceOrigin === 'null' || serviceOrigin.length <= 0 ){
+            serverCoreUrl = identityServerURL.getServerURL("", true, false);
+            serverUrl = identityServerURL.getServerURL("services", true, true);
+        } else {
+            serverCoreUrl = serviceOrigin + proxyContextPath;
+            serverUrl = serverCoreUrl + "/services";
```

Comment:
IMO, we should not append proxyContextPath for this configured serviceOrigin URL. That's expected to be configured with the exact path. 
This respective origin could point to the internal server as well, which conflicts when you append proxycontextPath derived from proxyhost and webcontext root